openedx/openedx-platform
# Fix whitespace issues in some capa problem index_dictionary content [FC-0062]

## Description

Capa problems support the `index_dictionary` API for adding their content to the search index. But currently, the text summary of the problem in the search index has some whitespace issues, especially with capa XML generated by the new MFE editors.

On the left you can see our new studio library search results page showing the `index_dictionary` text, and note that the "Option AOption B" is all run together. On the right, you can see the preview of the same problem.

![Screenshot showing the issue](https://github.com/user-attachments/assets/daa9c2a4-c273-49ae-8ca0-97d625d49f35)

After this fix:

![Screenshot showing the fixed version](https://github.com/user-attachments/assets/2b2667de-f735-4781-be8c-8a2c58ab9123)

## Supporting information

Part of https://github.com/openedx/frontend-app-course-authoring/issues/1091

Private ref: [FAL-3825](https://tasks.opencraft.com/browse/FAL-3825)

## Testing instructions

Should be well covered by the updated test case. But if you want to test manually:

1. Create a capa problem in a v2 library, view it in the authoring MFE, and note the problem is occurring.
2. Check out this PR then run `tutor dev run cms ./manage.py cms reindex_studio --experimental`
3. Refresh the library view and note the whitespace is improved.
